core/server/src/main/java/alluxio/master/file/meta/Inode.java

```diff
@@ -192,8 +193,22 @@ public T setGroup(String group) {
    * @return the updated object
    */
   public T setLastModificationTimeMs(long lastModificationTimeMs) {
-    mLastModificationTimeMs = lastModificationTimeMs;
-    return getThis();
+    return setLastModificationTimeMs(lastModificationTimeMs, false);
+  }
+
+  /**
+   * @param lastModificationTimeMs the last modification time to use
+   * @param override if true, sets the value regardless of the previous last modified time,
```

Comment:
When replaying from the journal, we create inodes which have default `lastModificationTimeMs` to be `System.currentTimeMillis` and then set the modification time to the correct time. If you prefer we could keep the old `setLastModificationTimeMs` and introduce a new method called `updateLastModificationTimeMs` which has the logic of only taking newer times.